configs/apps/ufs-srw-public-v2/spack.yaml

```diff
@@ -0,0 +1,64 @@
+spack:
+  packages:
+    zlib:
+      version: [1.2.11]
```

Comment:
Didn't we want to put those version informations into the `env` virtual packages? Or are these environments "special" enough to be treated differently? This would then apply to any release, or? WIth upcoming quarterly releases by EPIC and JCSDA, this will mean *a lot of* environments/apps. That's ok, but we should be really clear how we do the versioning so that it is consistent from spack-stack version to spack-stack version and we don't have to change it and the documentation.

Reply:
Hmmm, I think this is a case of a special standalone environment, and it should be handled in the `spack.yaml`. I think the apps could be cut down significantly to just special environments. Most of them just add a single spec to an otherwise empty `spack.yaml`. The `-env` specifies what to build and the environment applies the necessary special versions.

In case of future releases, once everything is using Spack it would hopefully be as easy as providing an existing tag rather than having these special versions. Or it could be in a `release/public-v2` branch.

Seems nicer to add it to an environment then adding each time.

```
    depends_on('netcdf-fortran@4.5.3', when='public-v2')
    depends_on('parallelio@2.5.2', when='public-v2')
    ...
```

Reply:
I was thinking of something like that:
```
    with when('@main'):
        if sys.platform == 'darwin':
            depends_on('libbacktrace', type='run')
        depends_on('cmake', type='run')
        depends_on('curl', type='run')
        depends_on('git', type='run')
        depends_on('hdf5', type='run')
        depends_on('nccmp', type='run')
        depends_on('netcdf-c', type='run')
        depends_on('netcdf-fortran', type='run')
        depends_on('parallel-netcdf', type='run')
        depends_on('parallelio', type='run')
        depends_on('python@3.7:')
        depends_on('szip', type='run')
        depends_on('wget', type='run')
        depends_on('zlib', type='run')

    with when('@1.0.0'):
        if sys.platform == 'darwin':
            depends_on('libbacktrace', type='run')
        depends_on('cmake@3.21:', type='run')
        depends_on('curl', type='run')
        depends_on('git', type='run')
        depends_on('hdf5@1.12.1', type='run')
        depends_on('nccmp@1.9.0.1', type='run')
        depends_on('netcdf-c@4.8.1', type='run')
        depends_on('netcdf-fortran@4.5.4', type='run')
        depends_on('parallel-netcdf@1.12.2', type='run')
        depends_on('parallelio@2.5.4', type='run')
        depends_on('python@3.7:')
        depends_on('szip', type='run')
        depends_on('wget', type='run')
        depends_on('zlib@1.2.11', type='run')
```

Reply:
I'll do the same for the jedi environments. The snippet I sent you is my suggestion for `base-env`, other than that we shouldn't have any overlap.

Reply:
A downside I see though is that you have to get real specific with low-level dependencies like `zlib` and `jasper` instead of just `depend_on(g2)` and let g2 handle the jasper dependency.

Reply:
I don't think we have to. I added the `zlib` version dependency, but I don't think it is required. We would put the versions that we really care of (`g2`, `netcdf-c`, ...) in these env files and let them handle their dependencies. In case there are special restrictions on versions, such as the need to use `pythran@0.11:` because earlier versions don't build, we can keep those in the common `packages.yaml`? This file would only contain versions for those special cases and the flavors for the packages. Or we get rid of that file entirely and have all information in the `env` files?

Reply:
I think it'll be fine for future releases because we'll have consistent versions that are already being used, but for the `release/public-v2` the `srw_common` contains special versions of zlib, jasper, libpng.